.github/workflows/no-cheat.yml

```diff
@@ -20,11 +20,14 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Verify no lint is disabled in the new code
+      - name: Verify no additional disabled lint in the new code
         run: |
-          NEW_CODE=$(git diff origin/main..$(git branch --show-current) | grep -e '^+')
-          CHEAT=$(echo "${NEW_CODE}" | grep '# pylint: disable' | grep -v "CHEAT" | wc -c)
-          if [ "${CHEAT}" -ne 0 ]; then
-            echo "Do not cheat the linter: ${CHEAT}"
+          MAIN_CODE=$(git diff origin/main..$(git branch --show-current) | grep -e '^-')
+          MAIN_CHEATS=$(echo "${MAIN_CODE}" | grep -c '# pylint: disable')
+          BRANCH_CODE=$(git diff origin/main..$(git branch --show-current) | grep -e '^+')
+          BRANCH_CHEATS=$(echo "${BRANCH_CODE}" | grep -c '# pylint: disable')
+          NEW_CHEATS=$((BRANCH_CHEATS-MAIN_CHEATS))
```

Comment:
this would be incorrect when 

```
- ... # pylint: disable=foo
+ ... # pylint: disable=bar
```

🤷‍♂️ probably comparing on $1 from `^.* # pylint: disable=([\w-]+)$` is sufficient then. please create a PR into this PR to demonstrate that this case works.

Reply:
I've moved the analysis to python

Reply:
PR #2154 uses this script successfully, see https://github.com/databrickslabs/ucx/actions/runs/9878593859/job/27282880063?pr=2154

Change in tests/integration/conftest.py triggers:
Do not cheat the linter: found 1 additional # pylint: disable=some-code

Changes in tests/unit/workspace_access/test_listing.py do not trigger a cheat

Other cheats are triggered by code in this PR